### PR TITLE
WIP: Improve make frameworks rule and remove codesigning options

### PR DESCRIPTION
**Pull request by jmoody**
**WIP WIP WIP**
### Motivation

This is a work in progress.  I don't know if this is correct approach to take.  It does resolve the codesigning problems and allows me to build the frameworks.

`make frameworks` fails because the FBControlCore signing expects a team that does not exist.

In Xcode 7, we can select 'None' as a code signing option - this PR uses none.

 @sapieneptus As I understand it, in Xcode 8, we must provide a team?  [Possible solution](https://github.com/calabash/calabash-codesign/pull/17).

I am not sure why we need "-" code sign option - I understand what it does, but do we actually need to codesign anything?  This PR removes that option from FBDeviceControl.

I updated the `bin/make/frameworks.sh` to use xcpretty if it is available and to fail fast if one of the build steps fails.

### CODE REVIEW

File: bin/make/frameworks.sh [hunk 1]
```diff
@@ -1,7 +1,58 @@
...
+  XC_PIPE='xcpretty -c'
+else
+  XC_PIPE='cat'
+fi
+
+BUILD_DIR="build"
+XC_PROJECT="FBSimulatorControl.xcodeproj"
+
+xcrun xcodebuild \
+  -SYMROOT="${BUILD_DIR}" \
```
Comment (sapieneptus) [inline bin/make/frameworks.sh:24]: @jmoody I probably would have just made this a function but I'm not going to block for a nitpick

**Comment by sapieneptus** [comment]
@jmoody What exactly is the 'WIP' part?

**Comment by jmoody** [comment]
These changes remove all code signing.  

I am not sure that is what we want to do.  It is not clear to me if anything needs to be codesigned.  What is your understand of the code signing requirements?

If we need a code signing team for Xcode 8, I suggest we use the Calabash.keychain and hard code the Mac Developer: Karl Krukow (XXXX) team. [reference](https://github.com/calabash/calabash-codesign/pull/17).

**Comment by jmoody** [comment]
@sapieneptus I am finished making changes.  This can be reviewed.

File: bin/make/frameworks.sh [hunk 2]
```diff
@@ -1,7 +1,58 @@
 #!/usr/bin/env bash
 
 rm -rf build
-xcodebuild -target FBSimulatorControl
-xcodebuild -target FBDeviceControl
-xcodebuild -target FBControlCore
-xcodebuild -target XCTestBootstrap
+
+set -e
+set -o pipefail
```
Comment (sapieneptus) [inline bin/make/frameworks.sh:6]: I did not know about this, good to know!

**Pull request merged**